In [18]:
import pandas as pd

In [19]:
df_fuentes = pd.read_csv("../data/madrid/fuentes/Inventario Fuentes2023.csv", sep=";")
df_fuentes

CODIGO ZONA VERDE / \r\nVIA PUBLICA DISTRITO  \
0         FUE_01_0001                           ZV   CENTRO   
1         FUE_01_0002                           ZV   CENTRO   
2         FUE_01_0003                           ZV   CENTRO   
3         FUE_01_0004                           ZV   CENTRO   
4         FUE_01_0005                           ZV   CENTRO   
...               ...                          ...      ...   
2144     FUEP_21_0035                           ZV  BARAJAS   
2145     FUEP_21_0036                           ZV  BARAJAS   
2146     FUEP_21_0037                           ZV  BARAJAS   
2147  FUEP_21_12_0038                           ZV  BARAJAS   
2148  FUEP_21_12_0039                           ZV  BARAJAS   

                 DIRECCION        COMPLEMENTO_DIRECCION     COORD, X  \
0       PLAZA COMENDADORAS           PLAZA COMENDADORAS  439892,4997   
1     JARDINES CONDE DUQUE         JARDINES CONDE DUQUE  439766,5391   
2     PLAZA DE LAS SALESAS         PLAZA DE LAS SALESAS  441077,2599   
3     PLAZA VILLA DE PARIS         PLAZA VILLA DE PARIS   441281,789   
4     PLAZA VILLA DE PARIS         PLAZA VILLA DE PARIS  441199,3988   
...                    ...                          ...          ...   
2144         JUAN CARLOS I             B.11 · PASEO SUR    448355,41   
2145         JUAN CARLOS I             B.11 · PASEO SUR    448118,84   
2146         JUAN CARLOS I             A.01 · PLAZA SUR    448247,84   
2147         JUAN CARLOS I             D.09 · TALUD RÍA    448359,97   
2148         JUAN CARLOS I  D.10 · OLIVAR 3 LA ANDALUZA    448411,63   

         COORD, Y  SISTEMA DE COORDENADAS  FECHA DATOS  longitud    latitud  
0     4475419,798  ETRS89 / UTM  Zona 30N        45013 -3.708600  40.427243  
1     4475585,204  ETRS89 / UTM  Zona 30N        45013 -3.710101  40.428724  
2     4475077,763  ETRS89 / UTM  Zona 30N        45013 -3.694603  40.424246  
3       4475192,9  ETRS89 / UTM  Zona 30N        45013 -3.692202  40.425298  
4     4475222,805  ETRS89 / UTM  Zona 30N        45013 -3.693176  40.425562  
...           ...                     ...          ...       ...        ...  
2144   4478638,03  ETRS89 / UTM  Zona 30N        45013 -3.609099  40.456803  
2145    4478829,1  ETRS89 / UTM  Zona 30N        45013 -3.611905  40.458510  
2146   4478883,13  ETRS89 / UTM  Zona 30N        45013 -3.610388  40.459004  
2147   4478982,41  ETRS89 / UTM  Zona 30N        45013 -3.609074  40.459906  
2148   4479100,72  ETRS89 / UTM  Zona 30N        45013 -3.608474  40.460975  

[2149 rows x 11 columns]

In [20]:
df_fuentes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2149 entries, 0 to 2148
Data columns (total 11 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   CODIGO                      2149 non-null   object 
 1   ZONA VERDE / 
VIA PUBLICA  2149 non-null   object 
 2   DISTRITO                    2149 non-null   object 
 3   DIRECCION                   2149 non-null   object 
 4   COMPLEMENTO_DIRECCION       1771 non-null   object 
 5   COORD, X                    2149 non-null   object 
 6   COORD, Y                    2149 non-null   object 
 7   SISTEMA DE COORDENADAS      2149 non-null   object 
 8   FECHA DATOS                 2149 non-null   int64  
 9   longitud                    2149 non-null   float64
 10  latitud                     2149 non-null   float64
dtypes: float64(2), int64(1), object(8)
memory usage: 184.8+ KB


In [21]:
df_fuentes.describe(include="all")

CODIGO ZONA VERDE / \r\nVIA PUBLICA         DISTRITO   DIRECCION  \
count       2149                         2149             2149        2149   
unique      2131                            2               24        1464   
top     PDTE CI6                           ZV  MONCLOA ARAVACA  MADRID RÍO   
freq          12                         1791              231          68   
mean         NaN                          NaN              NaN         NaN   
std          NaN                          NaN              NaN         NaN   
min          NaN                          NaN              NaN         NaN   
25%          NaN                          NaN              NaN         NaN   
50%          NaN                          NaN              NaN         NaN   
75%          NaN                          NaN              NaN         NaN   
max          NaN                          NaN              NaN         NaN   

       COMPLEMENTO_DIRECCION   COORD, X    COORD, Y  SISTEMA DE COORDENADAS  \
count                   1771       2149        2149                    2149   
unique                  1489       2137        2137                       1   
top      JUNTO ZONA INFANTIL  441750,78  4481010,06  ETRS89 / UTM  Zona 30N   
freq                      35          4           4                    2149   
mean                     NaN        NaN         NaN                     NaN   
std                      NaN        NaN         NaN                     NaN   
min                      NaN        NaN         NaN                     NaN   
25%                      NaN        NaN         NaN                     NaN   
50%                      NaN        NaN         NaN                     NaN   
75%                      NaN        NaN         NaN                     NaN   
max                      NaN        NaN         NaN                     NaN   

        FECHA DATOS     longitud      latitud  
count        2149.0  2149.000000  2149.000000  
unique          NaN          NaN          NaN  
top             NaN          NaN          NaN  
freq            NaN          NaN          NaN  
mean        45013.0    -3.679796    40.426464  
std             0.0     0.050000     0.040344  
min         45013.0    -3.831163    40.329043  
25%         45013.0    -3.713140    40.397394  
50%         45013.0    -3.683962    40.424743  
75%         45013.0    -3.639784    40.459417  
max         45013.0    -3.562745    40.592700

In [22]:
df_fuentes["COORD, X"] = df_fuentes["COORD, X"].str.replace(",",".")
df_fuentes["COORD, Y"] = df_fuentes["COORD, Y"].str.replace(",",".")
df_fuentes

CODIGO ZONA VERDE / \r\nVIA PUBLICA DISTRITO  \
0         FUE_01_0001                           ZV   CENTRO   
1         FUE_01_0002                           ZV   CENTRO   
2         FUE_01_0003                           ZV   CENTRO   
3         FUE_01_0004                           ZV   CENTRO   
4         FUE_01_0005                           ZV   CENTRO   
...               ...                          ...      ...   
2144     FUEP_21_0035                           ZV  BARAJAS   
2145     FUEP_21_0036                           ZV  BARAJAS   
2146     FUEP_21_0037                           ZV  BARAJAS   
2147  FUEP_21_12_0038                           ZV  BARAJAS   
2148  FUEP_21_12_0039                           ZV  BARAJAS   

                 DIRECCION        COMPLEMENTO_DIRECCION     COORD, X  \
0       PLAZA COMENDADORAS           PLAZA COMENDADORAS  439892.4997   
1     JARDINES CONDE DUQUE         JARDINES CONDE DUQUE  439766.5391   
2     PLAZA DE LAS SALESAS         PLAZA DE LAS SALESAS  441077.2599   
3     PLAZA VILLA DE PARIS         PLAZA VILLA DE PARIS   441281.789   
4     PLAZA VILLA DE PARIS         PLAZA VILLA DE PARIS  441199.3988   
...                    ...                          ...          ...   
2144         JUAN CARLOS I             B.11 · PASEO SUR    448355.41   
2145         JUAN CARLOS I             B.11 · PASEO SUR    448118.84   
2146         JUAN CARLOS I             A.01 · PLAZA SUR    448247.84   
2147         JUAN CARLOS I             D.09 · TALUD RÍA    448359.97   
2148         JUAN CARLOS I  D.10 · OLIVAR 3 LA ANDALUZA    448411.63   

         COORD, Y  SISTEMA DE COORDENADAS  FECHA DATOS  longitud    latitud  
0     4475419.798  ETRS89 / UTM  Zona 30N        45013 -3.708600  40.427243  
1     4475585.204  ETRS89 / UTM  Zona 30N        45013 -3.710101  40.428724  
2     4475077.763  ETRS89 / UTM  Zona 30N        45013 -3.694603  40.424246  
3       4475192.9  ETRS89 / UTM  Zona 30N        45013 -3.692202  40.425298  
4     4475222.805  ETRS89 / UTM  Zona 30N        45013 -3.693176  40.425562  
...           ...                     ...          ...       ...        ...  
2144   4478638.03  ETRS89 / UTM  Zona 30N        45013 -3.609099  40.456803  
2145    4478829.1  ETRS89 / UTM  Zona 30N        45013 -3.611905  40.458510  
2146   4478883.13  ETRS89 / UTM  Zona 30N        45013 -3.610388  40.459004  
2147   4478982.41  ETRS89 / UTM  Zona 30N        45013 -3.609074  40.459906  
2148   4479100.72  ETRS89 / UTM  Zona 30N        45013 -3.608474  40.460975  

[2149 rows x 11 columns]

In [133]:
# df_fuentes["COORD, X"] = df_fuentes["COORD, X"].astype(float)
# df_fuentes["COORD, Y"] = df_fuentes["COORD, Y"].astype(float)
# df_fuentes

No hemos podido pasar los datos a Float porque hay un campo tanto en coordenada X como en y con el texto *no aplica*. Vamos a filtrar aquellos campos en los que se indique *no aplica*.

In [23]:
df_fuentes[df_fuentes["COORD, X"] == 'no aplica']

CODIGO ZONA VERDE / \r\nVIA PUBLICA DISTRITO        DIRECCION  \
10  FUE_01_0014                           ZV   CENTRO  PASEO DEL PRADO   

   COMPLEMENTO_DIRECCION   COORD, X   COORD, Y  SISTEMA DE COORDENADAS  \
10       PASEO DEL PRADO  no aplica  no aplica  ETRS89 / UTM  Zona 30N   

    FECHA DATOS  longitud    latitud  
10        45013 -3.693159  40.414374

Hemos preguntado a ChatGPT como podemos obtener los datos de *COORD, X* y *COORD, Y*, obteniendo la siguiente respuesta:

Utilizando las coordenadas geográficas proporcionadas (longitud: -3.693159, latitud: 40.414374) y el sistema de coordenadas proyectadas "ETRS89 / UTM Zona 30N", la conversión da como resultado las siguientes coordenadas proyectadas COORDX y COORDY:

*COORD, X*: 441131.329643776

*COORD, Y*: 4473173.342050204

Se ha comprobado con otros datos en el que sí tenemos *COORD, X*, *COORD, Y*, *latitud* y *longitud*, que el resultado que da ChatGPT es correcto (es decir, le hemos pasado únicamente los datos de latitud y longitud y nos ha devuelto el mismo resultado de *COORD, X* y *COORD, Y* que está presente en los datos).

In [24]:
df_fuentes["COORD, X"] = df_fuentes["COORD, X"].str.replace("no aplica","441131.329643776")
df_fuentes["COORD, Y"] = df_fuentes["COORD, Y"].str.replace("no aplica","4473173.342050204")

Una vez aplicado el cambio, ya podemos pasar los datos a tipo float.

In [25]:
df_fuentes["COORD, X"] = df_fuentes["COORD, X"].astype(float)
df_fuentes["COORD, Y"] = df_fuentes["COORD, Y"].astype(float)
df_fuentes

CODIGO ZONA VERDE / \r\nVIA PUBLICA DISTRITO  \
0         FUE_01_0001                           ZV   CENTRO   
1         FUE_01_0002                           ZV   CENTRO   
2         FUE_01_0003                           ZV   CENTRO   
3         FUE_01_0004                           ZV   CENTRO   
4         FUE_01_0005                           ZV   CENTRO   
...               ...                          ...      ...   
2144     FUEP_21_0035                           ZV  BARAJAS   
2145     FUEP_21_0036                           ZV  BARAJAS   
2146     FUEP_21_0037                           ZV  BARAJAS   
2147  FUEP_21_12_0038                           ZV  BARAJAS   
2148  FUEP_21_12_0039                           ZV  BARAJAS   

                 DIRECCION        COMPLEMENTO_DIRECCION     COORD, X  \
0       PLAZA COMENDADORAS           PLAZA COMENDADORAS  439892.4997   
1     JARDINES CONDE DUQUE         JARDINES CONDE DUQUE  439766.5391   
2     PLAZA DE LAS SALESAS         PLAZA DE LAS SALESAS  441077.2599   
3     PLAZA VILLA DE PARIS         PLAZA VILLA DE PARIS  441281.7890   
4     PLAZA VILLA DE PARIS         PLAZA VILLA DE PARIS  441199.3988   
...                    ...                          ...          ...   
2144         JUAN CARLOS I             B.11 · PASEO SUR  448355.4100   
2145         JUAN CARLOS I             B.11 · PASEO SUR  448118.8400   
2146         JUAN CARLOS I             A.01 · PLAZA SUR  448247.8400   
2147         JUAN CARLOS I             D.09 · TALUD RÍA  448359.9700   
2148         JUAN CARLOS I  D.10 · OLIVAR 3 LA ANDALUZA  448411.6300   

         COORD, Y  SISTEMA DE COORDENADAS  FECHA DATOS  longitud    latitud  
0     4475419.798  ETRS89 / UTM  Zona 30N        45013 -3.708600  40.427243  
1     4475585.204  ETRS89 / UTM  Zona 30N        45013 -3.710101  40.428724  
2     4475077.763  ETRS89 / UTM  Zona 30N        45013 -3.694603  40.424246  
3     4475192.900  ETRS89 / UTM  Zona 30N        45013 -3.692202  40.425298  
4     4475222.805  ETRS89 / UTM  Zona 30N        45013 -3.693176  40.425562  
...           ...                     ...          ...       ...        ...  
2144  4478638.030  ETRS89 / UTM  Zona 30N        45013 -3.609099  40.456803  
2145  4478829.100  ETRS89 / UTM  Zona 30N        45013 -3.611905  40.458510  
2146  4478883.130  ETRS89 / UTM  Zona 30N        45013 -3.610388  40.459004  
2147  4478982.410  ETRS89 / UTM  Zona 30N        45013 -3.609074  40.459906  
2148  4479100.720  ETRS89 / UTM  Zona 30N        45013 -3.608474  40.460975  

[2149 rows x 11 columns]

A continuación, eliminamos las columnas de *FECHA DATOS* y *SISTEMA DE COORDENADAS*. La primera porque la fecha no es interesante para nuestro proceso y la segunda porque solo existe un valor.

In [26]:
df_fuentes.drop(["FECHA DATOS","SISTEMA DE COORDENADAS"],axis=1,inplace=True)
df_fuentes

CODIGO ZONA VERDE / \r\nVIA PUBLICA DISTRITO  \
0         FUE_01_0001                           ZV   CENTRO   
1         FUE_01_0002                           ZV   CENTRO   
2         FUE_01_0003                           ZV   CENTRO   
3         FUE_01_0004                           ZV   CENTRO   
4         FUE_01_0005                           ZV   CENTRO   
...               ...                          ...      ...   
2144     FUEP_21_0035                           ZV  BARAJAS   
2145     FUEP_21_0036                           ZV  BARAJAS   
2146     FUEP_21_0037                           ZV  BARAJAS   
2147  FUEP_21_12_0038                           ZV  BARAJAS   
2148  FUEP_21_12_0039                           ZV  BARAJAS   

                 DIRECCION        COMPLEMENTO_DIRECCION     COORD, X  \
0       PLAZA COMENDADORAS           PLAZA COMENDADORAS  439892.4997   
1     JARDINES CONDE DUQUE         JARDINES CONDE DUQUE  439766.5391   
2     PLAZA DE LAS SALESAS         PLAZA DE LAS SALESAS  441077.2599   
3     PLAZA VILLA DE PARIS         PLAZA VILLA DE PARIS  441281.7890   
4     PLAZA VILLA DE PARIS         PLAZA VILLA DE PARIS  441199.3988   
...                    ...                          ...          ...   
2144         JUAN CARLOS I             B.11 · PASEO SUR  448355.4100   
2145         JUAN CARLOS I             B.11 · PASEO SUR  448118.8400   
2146         JUAN CARLOS I             A.01 · PLAZA SUR  448247.8400   
2147         JUAN CARLOS I             D.09 · TALUD RÍA  448359.9700   
2148         JUAN CARLOS I  D.10 · OLIVAR 3 LA ANDALUZA  448411.6300   

         COORD, Y  longitud    latitud  
0     4475419.798 -3.708600  40.427243  
1     4475585.204 -3.710101  40.428724  
2     4475077.763 -3.694603  40.424246  
3     4475192.900 -3.692202  40.425298  
4     4475222.805 -3.693176  40.425562  
...           ...       ...        ...  
2144  4478638.030 -3.609099  40.456803  
2145  4478829.100 -3.611905  40.458510  
2146  4478883.130 -3.610388  40.459004  
2147  4478982.410 -3.609074  40.459906  
2148  4479100.720 -3.608474  40.460975  

[2149 rows x 9 columns]

In [27]:
df_fuentes[df_fuentes["CODIGO"].duplicated()]

CODIGO ZONA VERDE / \r\nVIA PUBLICA           DISTRITO  \
188         SKATEPARK                           ZV         ARGANZUELA   
189         SKATEPARK                           ZV         ARGANZUELA   
190         SKATEPARK                           ZV         ARGANZUELA   
191         SKATEPARK                           ZV         ARGANZUELA   
811              #N/D                           ZV    MONCLOA ARAVACA   
815              #N/D                           ZV    MONCLOA ARAVACA   
894          PDTE CI6                           ZV    MONCLOA ARAVACA   
895          PDTE CI6                           ZV    MONCLOA ARAVACA   
896          PDTE CI6                           ZV    MONCLOA ARAVACA   
897          PDTE CI6                           ZV    MONCLOA ARAVACA   
1684  FUEP_16_15_0017                           ZV          HORTALEZA   
1853         PDTE CI6                           ZV  VILLA DE VALLECAS   
1854         PDTE CI6                           ZV  VILLA DE VALLECAS   
1855         PDTE CI6                           ZV  VILLA DE VALLECAS   
1856         PDTE CI6                           ZV  VILLA DE VALLECAS   
1857         PDTE CI6                           ZV  VILLA DE VALLECAS   
1858         PDTE CI6                           ZV  VILLA DE VALLECAS   
1859         PDTE CI6                           ZV  VILLA DE VALLECAS   

                                           DIRECCION  \
188                                       MADRID RÍO   
189                                       MADRID RÍO   
190                                       MADRID RÍO   
191                                       MADRID RÍO   
811                      CASA DE CAMPO ZONA FORESTAL   
815                        CASA DE CAMPO ZONA URBANA   
894                                      PZA. ESPAÑA   
895                                      PZA. ESPAÑA   
896                                      PZA. ESPAÑA   
897                                      PZA. ESPAÑA   
1684  PARQUE FORESTAL DE VALDEBEBAS-PARQUE FELIPE VI   
1853                              Parque de La Gavia   
1854                              Parque de La Gavia   
1855                              Parque de La Gavia   
1856                              Parque de La Gavia   
1857                              Parque de La Gavia   
1858                              Parque de La Gavia   
1859                              Parque de La Gavia   

                                  COMPLEMENTO_DIRECCION     COORD, X  \
188                Z.21. - ARGANZUELA - PUENTE DE PRAGA  440252.7000   
189                Z.21. - ARGANZUELA - PUENTE DE PRAGA  440253.5000   
190                       Z.18. - ARGANZUELA - PERRAULT  439699.0000   
191                       Z.18. - ARGANZUELA - PERRAULT  439831.7000   
811   FUENTE DEL APARCAMIENTO DE RODAJOS (JUNTO A JU...  434004.5836   
815    FUENTE DEL SOTERRAMIENTO DE LA LINEA 10 DE METRO  434981.3176   
894            zona calle Irún, junto estatua 2 de mayo  439367.7200   
895                       áreas infantiles Plaza España  439476.4700   
896                       áreas infantiles Plaza España  439513.5600   
897                       áreas infantiles Plaza España  439530.0600   
1684                                                NaN  445728.8800   
1853                                     Plaza del agua  447107.8600   
1854                             Bosque de los sentidos  447076.8400   
1855                             Bosque de los sentidos  446742.6800   
1856                                     Zona deportiva  446670.9800   
1857           Avenida del Mayorazgo/ Parkour-Rocodromo  446642.5200   
1858                               Colina del reciclaje  446611.2900   
1859                                        Área canina  446592.4700   

         COORD, Y  longitud    latitud  
188   4471888.900 -3.704023  40.395461  
189   4471888.500 -3.704013  40.395457  
190   4472320.800 -3.710587  40.399312  
191   4472229.900 -3.709015  

Dejamos los duplicados porque tienen distintas longitudes y latitudes, por lo que podría ser que existan varias fuentes en un entorno cercano que dispongan del mismo código. Todos los códigos duplicados se encuentran en zonas verdes.

In [28]:
df_fuentes["COMPLEMENTO_DIRECCION"].fillna("Sin información adicional",inplace=True)

Se completan los valores nulos de la columna *COMPLEMENTO_DIRECCION* con el siguiente mensaje *Sin información adicional*

In [29]:
df_fuentes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2149 entries, 0 to 2148
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   CODIGO                      2149 non-null   object 
 1   ZONA VERDE / 
VIA PUBLICA  2149 non-null   object 
 2   DISTRITO                    2149 non-null   object 
 3   DIRECCION                   2149 non-null   object 
 4   COMPLEMENTO_DIRECCION       2149 non-null   object 
 5   COORD, X                    2149 non-null   float64
 6   COORD, Y                    2149 non-null   float64
 7   longitud                    2149 non-null   float64
 8   latitud                     2149 non-null   float64
dtypes: float64(4), object(5)
memory usage: 151.2+ KB


In [30]:
df_fuentes.describe(include="all")

CODIGO ZONA VERDE / \r\nVIA PUBLICA         DISTRITO   DIRECCION  \
count       2149                         2149             2149        2149   
unique      2131                            2               24        1464   
top     PDTE CI6                           ZV  MONCLOA ARAVACA  MADRID RÍO   
freq          12                         1791              231          68   
mean         NaN                          NaN              NaN         NaN   
std          NaN                          NaN              NaN         NaN   
min          NaN                          NaN              NaN         NaN   
25%          NaN                          NaN              NaN         NaN   
50%          NaN                          NaN              NaN         NaN   
75%          NaN                          NaN              NaN         NaN   
max          NaN                          NaN              NaN         NaN   

            COMPLEMENTO_DIRECCION      COORD, X      COORD, Y     longitud  \
count                        2149  2.149000e+03  2.149000e+03  2149.000000   
unique                       1490           NaN           NaN          NaN   
top     Sin información adicional           NaN           NaN          NaN   
freq                          378           NaN           NaN          NaN   
mean                          NaN  4.966480e+06  4.814026e+07    -3.679796   
std                           NaN  4.449661e+07  4.396529e+08     0.050000   
min                           NaN  4.295447e+05  4.464488e+06    -3.831163   
25%                           NaN  4.395039e+05  4.472121e+06    -3.713140   
50%                           NaN  4.420196e+05  4.475193e+06    -3.683962   
75%                           NaN  4.458438e+05  4.479170e+06    -3.639784   
max                           NaN  4.486905e+08  4.481422e+09    -3.562745   

            latitud  
count   2149.000000  
unique          NaN  
top             NaN  
freq            NaN  
mean      40.426464  
std        0.040344  
min       40.329043  
25%       40.397394  
50%       40.424743  
75%       40.459417  
max       40.592700

In [31]:
df_fuentes["ZONA VERDE / \r\nVIA PUBLICA"].unique()

array(['ZV', 'VP'], dtype=object)

In [32]:
df_fuentes["DISTRITO"].unique()

array(['CENTRO', 'ARGANZUELA', 'RETIRO', 'SALAMANCA', 'CHAMARTIN',
       'TETUAN', 'CHAMBERI', 'FUENCARRAL EL PARDO', 'FUENCARRAL-EL PARDO',
       'MONCLOA ARAVACA', 'LATINA', 'CARABANCHEL', 'USERA',
       'PUENTE DE VALLECAS', 'PUENTE VALLECAS', 'MORATALAZ',
       'CIUDAD LINEAL', 'HORTALEZA', '16. Hortaleza', 'VILLAVERDE',
       'VILLA DE VALLECAS', 'VICALVARO', 'SAN BLAS', 'BARAJAS'],
      dtype=object)

In [34]:
df_fuentes = df_fuentes.loc[df_fuentes["DISTRITO"].isin(['CENTRO', 'CHAMBERI', 'RETIRO','ARGANZUELA','SALAMANCA'])]

Se ha observado que en la columna código hay algunos datos que contienen el siguiente string: *#N/D*

Se han reemplazado por el valor *No disponible*

In [35]:
df_fuentes[df_fuentes["CODIGO"] == "#N/D"]

Empty DataFrame
Columns: [CODIGO, ZONA VERDE / 
VIA PUBLICA, DISTRITO, DIRECCION, COMPLEMENTO_DIRECCION, COORD, X, COORD, Y, longitud, latitud]
Index: []

In [36]:
df_fuentes["CODIGO"] = df_fuentes["CODIGO"].str.replace("#N/D","No disponible")

C:\Users\usuario\AppData\Local\Temp\ipykernel_4992\3055129809.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_fuentes["CODIGO"] = df_fuentes["CODIGO"].str.replace("#N/D","No disponible")


In [37]:
df_fuentes.info()

<class 'pandas.core.frame.DataFrame'>
Index: 365 entries, 0 to 498
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   CODIGO                      365 non-null    object 
 1   ZONA VERDE / 
VIA PUBLICA  365 non-null    object 
 2   DISTRITO                    365 non-null    object 
 3   DIRECCION                   365 non-null    object 
 4   COMPLEMENTO_DIRECCION       365 non-null    object 
 5   COORD, X                    365 non-null    float64
 6   COORD, Y                    365 non-null    float64
 7   longitud                    365 non-null    float64
 8   latitud                     365 non-null    float64
dtypes: float64(4), object(5)
memory usage: 28.5+ KB


In [38]:
df_fuentes.describe()

COORD, X      COORD, Y    longitud     latitud
count  3.650000e+02  3.650000e+02  365.000000  365.000000
mean   1.648308e+06  1.672185e+07   -3.691932   40.415787
std    2.306001e+07  2.339923e+08    0.017076    0.014121
min    4.356313e+05  4.470642e+06   -3.758552   40.384318
25%    4.400085e+05  4.472937e+06   -3.707134   40.404984
50%    4.411298e+05  4.474155e+06   -3.693865   40.415855
75%    4.424144e+05  4.475331e+06   -3.678652   40.426554
max    4.410022e+08  4.474891e+09   -3.659918   40.446551

In [39]:
df_fuentes.describe(include="object")

CODIGO ZONA VERDE / \r\nVIA PUBLICA DISTRITO  \
count         365                          365      365   
unique        361                            2        5   
top     SKATEPARK                           ZV   CENTRO   
freq            5                          251      112   

                       DIRECCION      COMPLEMENTO_DIRECCION  
count                        365                        365  
unique                       266                        250  
top     JARDINES DEL BUEN RETIRO  Sin información adicional  
freq                          31                         52

In [149]:
# nombre_archivo = "../Json/fuentes.json"

In [150]:
# df_fuentes.to_json(nombre_archivo, orient='records')

### Nuevo enfoque, quieren todo en un .json, unificaremos columnas y los propios json en cada cosa. Solo 4 columnas tipo,nombre,direccion,coordenada-x,coordenada-y latitud,longitud y descripcion

In [40]:
df_fuentes.columns

Index(['CODIGO', 'ZONA VERDE / \r\nVIA PUBLICA', 'DISTRITO', 'DIRECCION',
       'COMPLEMENTO_DIRECCION', 'COORD, X', 'COORD, Y', 'longitud', 'latitud'],
      dtype='object')

In [41]:
def concatenar_fila(row):
    return f"{row['DISTRITO']}, {row['DIRECCION']}"

# Crear la nueva columna "DIRECCION" aplicando la función a lo largo de las filas
df_fuentes['DIRECCION'] = df_fuentes.apply(concatenar_fila, axis=1)

C:\Users\usuario\AppData\Local\Temp\ipykernel_4992\2047624356.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_fuentes['DIRECCION'] = df_fuentes.apply(concatenar_fila, axis=1)


In [42]:
df_fuentes["DIRECCION"][5]

'CENTRO, PLAZA CRISTINO MARTOS'

In [43]:
df_fuentes.columns

Index(['CODIGO', 'ZONA VERDE / \r\nVIA PUBLICA', 'DISTRITO', 'DIRECCION',
       'COMPLEMENTO_DIRECCION', 'COORD, X', 'COORD, Y', 'longitud', 'latitud'],
      dtype='object')

In [44]:
df_fuentes.drop(['ZONA VERDE / \r\nVIA PUBLICA', 'DISTRITO','COORD, X', 'COORD, Y'],axis = 1,inplace=True)

C:\Users\usuario\AppData\Local\Temp\ipykernel_4992\4069181178.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_fuentes.drop(['ZONA VERDE / \r\nVIA PUBLICA', 'DISTRITO','COORD, X', 'COORD, Y'],axis = 1,inplace=True)


In [45]:
df_fuentes.columns

Index(['CODIGO', 'DIRECCION', 'COMPLEMENTO_DIRECCION', 'longitud', 'latitud'], dtype='object')

In [46]:
df_fuentes["TIPO"] = "FUENTES"

C:\Users\usuario\AppData\Local\Temp\ipykernel_4992\1797101562.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_fuentes["TIPO"] = "FUENTES"


In [47]:
df_fuentes.rename(columns={'CODIGO': 'NOMBRE','longitud':'LONGITUD', 'latitud':'LATITUD',"COMPLEMENTO_DIRECCION":"DESCRIPCION"}, inplace=True)

C:\Users\usuario\AppData\Local\Temp\ipykernel_4992\479434510.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_fuentes.rename(columns={'CODIGO': 'NOMBRE','longitud':'LONGITUD', 'latitud':'LATITUD',"COMPLEMENTO_DIRECCION":"DESCRIPCION"}, inplace=True)


In [48]:
nuevo_orden_columnas = ["TIPO","NOMBRE","DIRECCION","DESCRIPCION","LATITUD","LONGITUD"] 
df_fuentes = df_fuentes.reindex(columns=nuevo_orden_columnas)

In [49]:
df_fuentes.head()

TIPO       NOMBRE                     DIRECCION           DESCRIPCION  \
0  FUENTES  FUE_01_0001    CENTRO, PLAZA COMENDADORAS    PLAZA COMENDADORAS   
1  FUENTES  FUE_01_0002  CENTRO, JARDINES CONDE DUQUE  JARDINES CONDE DUQUE   
2  FUENTES  FUE_01_0003  CENTRO, PLAZA DE LAS SALESAS  PLAZA DE LAS SALESAS   
3  FUENTES  FUE_01_0004  CENTRO, PLAZA VILLA DE PARIS  PLAZA VILLA DE PARIS   
4  FUENTES  FUE_01_0005  CENTRO, PLAZA VILLA DE PARIS  PLAZA VILLA DE PARIS   

     LATITUD  LONGITUD  
0  40.427243 -3.708600  
1  40.428724 -3.710101  
2  40.424246 -3.694603  
3  40.425298 -3.692202  
4  40.425562 -3.693176

In [50]:
df_fuentes.to_csv("../data/madrid/fuentes/fuentes_nuevo_reduced.csv", index=False)